src/main/java/com/WebController/EventController.java

```diff
@@ -51,16 +74,72 @@ public String receivePatient (@RequestBody List<Event> events){
                 eventRepo.save(event);
             }
         }
-        return "success";
+        return new ResponseEntity<>("saved", HttpStatus.OK);
     }
 
+    /**
+     * DoGet handler for all events.
+     * User will get a list of all existing events with number of attended patients
+     */
     @GetMapping(value = "/allEvents", produces = "application/json")
     @ResponseBody
-    public List<Event> getAllEvents (){
-        List<Event> list = new ArrayList<>();
-        Iterable<Event> events = this.eventRepo.findAll();
+    public Iterable<Event> getAllEvents (@RequestParam("page") int pageNum, ServletResponse response){
+        Iterable<Event> events = this.eventRepo.findAll( PageRequest.of(pageNum-1, 5));
+        Iterable<Event> nextPage = this.eventRepo.findAll( PageRequest.of(pageNum, 5));
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+        if (IterableUtil.sizeOf(nextPage)>0){ //have next page
+            httpServletResponse.addHeader("has-next-page", "true");
+        } else {
+            httpServletResponse.addHeader("has-next-page", "false");
+        }
+        List<Event> list=new ArrayList<>();
```

Comment:
You shouldn't need to convert the Iterable into a List. Just return the Iterable.